OpenOversight/app/commands.py

```diff
@@ -411,6 +421,388 @@ def bulk_add_officers(filename, no_create, update_by_name, update_static_fields)
         return len(ImportLog.created_officers), len(ImportLog.updated_officers)
 
 
+def _create_or_update_model(
+    row, existing_model_lookup, create_method, update_method, force_create=False, model=None
+):
+    if not row["id"]:
+        return create_method(row)
+    else:
+        if not force_create:
+            return update_method(row, existing_model_lookup[int(row["id"])])
+        else:
+            if model is not None:
+                existing = model.query.filter_by(id=int(row["id"])).first()
+                if existing:
+                    db.session.delete(existing)
+                    db.session.flush()
+            return create_method(row, force_id=True)
+
+
+def _check_provided_fields(dict_reader, required_fields, optional_fields, csv_name):
+    missing_required = set(required_fields) - set(dict_reader.fieldnames)
+    if len(missing_required) > 0:
+        raise Exception(
+            "Missing mandatory field(s) {} in {} csv.".format(
+                list(missing_required), csv_name
+            )
+        )
+    unexpected_fields = set(dict_reader.fieldnames) - set(
+        required_fields + optional_fields
+    )
+    if len(unexpected_fields) > 0:
+        raise Exception(
+            "Received unexpected field(s) {} in {} csv.".format(
+                list(unexpected_fields), csv_name
+            )
+        )
+
+
+def _objects_from_split_field(field, model_lookup):
+    if field:
+        return [model_lookup[object_id] for object_id in field.split("|")]
+    return []
+
+
+def _unify_field_names(fieldnames):
+    return [field_name.lower().replace(" ", "_") for field_name in fieldnames]
```

Comment:
I've used `inflection` to convert between cases before - not sure that adding a dep is better or worse than using a common-enough function, here.
https://inflection.readthedocs.io/en/latest/

Reply:
I didn't know that library, thanks! However I wasn't able to find a command that does what I was doing here and I also value that it doesn't require looking up a library to find out what it is happening here